File: tmdb-core/src/main/java/dev/reuss/tmdb/domain/tv/series/model/TvSeriesAggregateCrewCredit.java
package dev.reuss.tmdb.domain.tv.series.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import dev.reuss.tmdb.common.TmdbModel;
import dev.reuss.tmdb.util.TmdbCollections;

import java.util.List;

/**
 * Aggregate crew credit for a TMDB TV series.
 *
 * @param adult              whether the person is marked as adult
 * @param gender             TMDB gender value
 * @param id                 TMDB person id
 * @param knownForDepartment known department
 * @param name               person name
 * @param originalName       original person name
 * @param popularity         person popularity
 * @param profilePath        profile image path
 * @param jobs               aggregate crew jobs
 * @param department         crew department
 * @param totalEpisodeCount  total episode count
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record TvSeriesAggregateCrewCredit(
        boolean adult,

        int gender,

        int id,

        @JsonProperty("known_for_department")
        String knownForDepartment,

        String name,

        @JsonProperty("original_name")
        String originalName,

        double popularity,

        @JsonProperty("profile_path")
        String profilePath,

        List<TvSeriesAggregateCrewJob> jobs,

        String department,

        @JsonProperty("total_episode_count")
        int totalEpisodeCount
) implements TmdbModel {

    public TvSeriesAggregateCrewCredit {
        jobs = TmdbCollections.list(jobs);
    }
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/domain/tv/series/model/TvSeriesAggregateCrewCredit.kt
package dev.reuss.tmdb.domain.tv.series.model

import com.fasterxml.jackson.annotation.JsonProperty
import dev.reuss.tmdb.common.TmdbModel

/**
 * Aggregate crew credit for a TMDB TV series.
 *
 * @property adult whether the person is marked as adult
 * @property gender TMDB gender value
 * @property id TMDB person id
 * @property knownForDepartment known department
 * @property name person name
 * @property originalName original person name
 * @property popularity person popularity
 * @property profilePath profile image path
 * @property jobs aggregate crew jobs
 * @property department crew department
 * @property totalEpisodeCount total episode count
 */
@JvmRecord
data class TvSeriesAggregateCrewCredit(
    @all:JsonProperty("adult")
    val adult: Boolean,

    @all:JsonProperty("gender")
    val gender: Int,

    @all:JsonProperty("id")
    val id: Int,

    @all:JsonProperty("known_for_department")
    val knownForDepartment: String?,

    @all:JsonProperty("name")
    val name: String?,

    @all:JsonProperty("original_name")
    val originalName: String?,

    @all:JsonProperty("popularity")
    val popularity: Double,

    @all:JsonProperty("profile_path")
    val profilePath: String?,

    @all:JsonProperty("jobs")
    val jobs: List<TvSeriesAggregateCrewJob> = emptyList(),

    @all:JsonProperty("department")
    val department: String?,

    @all:JsonProperty("total_episode_count")
    val totalEpisodeCount: Int
) : TmdbModel